components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -1321,64 +1262,102 @@ export function createWalletApi () {
-                          }
-                        ]
-                      : [
-                          {
-                            accountId: arg.accountId,
-                            coin: arg.coin,
-                            chainId: arg.chainId,
-                            tokens: getEntitiesListFromEntityState(
-                              userTokens,
-                              userTokens.idsByChainId[
-                                networkEntityAdapter.selectId({
-                                  coin: arg.coin,
-                                  chainId: arg.chainId
-                                })
-                              ]
-                            )
-                          }
-                        ],
-                    {
-                      forceRefetch: true
+            const tokenBalancesRegistryArray = await mapLimit(
+              args.accounts,
+              Math.min(args.accounts.length, 3),
+              async (accountId: BraveWallet.AccountId) => {
+                const networks = args.networks.filter(
+                  (network) => network.coin === accountId.coin)
+
+                if (networks.length === 0) {
+                  return {}
+                }
+
+                const registryArray = await mapLimit(
+                  networks,
+                  Math.min(networks.length, 3),
+                  async (
+                    network: Pick<
+                      BraveWallet.NetworkInfo,
+                      'coin' | 'chainId'
+                    >
+                  ) => {
+                    const partialRegistryQuery = dispatch(
+                      getTokenBalancesForChainId.initiate(
+                        network.coin === CoinTypes.SOL
+                          ? [
+                              {
+                                accountId,
+                                coin: CoinTypes.SOL,
+                                chainId: network.chainId
+                              }
+                            ]
+                          : [
+                              {
+                                accountId,
+                                coin: coinTypesMapping[network.coin],
+                                chainId: network.chainId,
+                                tokens:
+                                  getEntitiesListFromEntityState(
+                                    userTokens,
+                                    userTokens.idsByChainId[
+                                      networkEntityAdapter.selectId({
+                                        coin: network.coin,
+                                        chainId: network.chainId
+                                      })
+                                    ]
+                                  )
+                              }
+                            ].filter(
+                              ({ coin }) => coin !== undefined
+                            ),
```

Comment:
How about this so that we don't have to call the filter function?
```suggestion
                          : coinTypesMapping[network.coin] ? [
                              {
                                accountId,
                                coin: coinTypesMapping[network.coin],
                                chainId: network.chainId,
                                tokens:
                                  getEntitiesListFromEntityState(
                                    userTokens,
                                    userTokens.idsByChainId[
                                      networkEntityAdapter.selectId({
                                        coin: network.coin,
                                        chainId: network.chainId
                                      })
                                    ]
                                  )
                              }
                            ] : [],
```